### PR TITLE
Nc longevity test

### PR DESCRIPTION
**Pull request by schneidy**
Adds in the bill and vote archive for NC. Bills go back to 1985 and votes go back to 1997.

### CODE REVIEW

File: openstates/nc/__init__.py [hunk 1]
```diff
@@ -17,6 +17,201 @@ class NorthCarolina(Jurisdiction):
...
+            "classification": "special",
+            "identifier": "2003E2",
+            "name": "2003 Extra Session on Economic Development Issues",
+            "start_date": "2003-12-09"
+        },
+        {
+            "_scraped_name": "2004 Extra Session on Economic Development Issues",
+            "classification": "special",
+            "identifier": "2003E3",
+            "name": "2004 Extra Session on Eco  nomic Development Issues",
```
Comment (jamesturk) [inline openstates/nc/__init__.py]: space is a typo

File: openstates/nc/bills.py [hunk 2]
```diff
@@ -70,10 +75,13 @@ def scrape_bill(self, chamber, session, bill_id):
             bill_type = "bill"
             bill_id = bill_id[0] + "B " + bill_id[1:]
 
-        bill_title = doc.xpath(
-            '/html/body/div/div/main/div[2]/div[contains(@class,"col-12")]/a'
-        )[0]
+        bill_title = doc.xpath("/html[1]/body[1]/div[1]/div[1]/main[1]/div[2]/div[1]")[
```
Comment (jamesturk) [inline openstates/nc/bills.py]: can you elaborate on this change?  does this work for all session as-is?

File: openstates/nc/bills.py [hunk 3]
```diff
@@ -70,10 +75,13 @@ def scrape_bill(self, chamber, session, bill_id):
...
+            0
+        ]
         bill_title = bill_title.text_content().strip()
+        if not bill_title:
```
Comment (jamesturk) [inline openstates/nc/bills.py]: is this supposed to be here? looks like debugging stuff that won't actually get called?

File: openstates/nc/bills.py [hunk 4]
```diff
@@ -159,7 +168,24 @@ def scrape_bill(self, chamber, session, bill_id):
...
+            if act_date is None:
+                search_action_date = action.split()
+                for act in search_action_date:
+                    try:
+                        if "/" in act:
+                            # try:
+                            act_date = dt.datetime.strptime(act, "%m/%d/%Y").strftime(
+                                "%Y-%m-%d"
+                            )
+                            #     print(type(act_date))
```
Comment (jamesturk) [inline openstates/nc/bills.py]: debugging stuff- remove?

File: openstates/nc/bills.py [hunk 5]
```diff
@@ -173,13 +199,17 @@ def scrape_bill(self, chamber, session, bill_id):
                     break
             else:
                 atype = None
-
-            bill.add_action(action, act_date, chamber=actor, classification=atype)
+            if act_date is not None:
```
Comment (jamesturk) [inline openstates/nc/bills.py]: if we're skipping actions without dates, let's comment why/how often that happens

File: openstates/nc/bills.py [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
 import lxml.html
+import re
 from pupa.scrape import Bill, Scraper, VoteEvent
 import pytz
+from collections import defaultdict
+
 
 eastern = pytz.timezone("US/Eastern")
 
+archived_votes = defaultdict(lambda: defaultdict(list))
```
Comment (jamesturk) [inline openstates/nc/bills.py]: can you move this to self.archived_votes

File: openstates/nc/bills.py [hunk 7]
```diff
@@ -258,14 +288,172 @@ def scrape_votes(self, bill, doc):
...
         chambers = [chamber] if chamber else ["upper", "lower"]
+
+        if session in ["1997", "1999"]:
+            self.scrape_archived_votes("upper", session)
+            self.scrape_archived_votes("lower", session)
+
         for chamber in chambers:
-            yield from self.scrape_chamber(chamber, session)
+
+            if session in ["1997", "1999"]:
```
Comment (jamesturk) [inline openstates/nc/bills.py]: this if statement doesn't do anything anymore :)

File: openstates/nc/bills.py (hunk 3 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Yes. It is for a few special cases where there is nothing in bill_title. It then creates a bill_title using the bill_id.

File: openstates/nc/bills.py (hunk 2 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Not all of the bill titles are within a link. The x-path works for all sessions.

File: openstates/nc/bills.py (hunk 3 above)
Comment (jamesturk) [reply openstates/nc/bills.py]: ah, can you add a comment explaining this & drop the print statement 👍

File: openstates/nc/__init__.py (hunk 1 above)
Comment (schneidy) [reply openstates/nc/__init__.py]: Fixed!

File: openstates/nc/bills.py (hunk 4 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Removed all print statements!

File: openstates/nc/bills.py (hunk 5 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Looking into how often this takes place. Honestly don't remember.

File: openstates/nc/bills.py (hunk 7 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Removed!

File: openstates/nc/bills.py (hunk 6 above)
Comment (schneidy) [reply openstates/nc/bills.py]: Moved!

**Pull request merged**